File: Day76/Part3/stagger_drain.cpp
#include <iostream>
#include <vector>

using namespace std;

int solve() {
    int n;
    long long T;
    if (!(cin >> n >> T)) return 0;
    
    vector<long long> a(n);
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }
    
    int stuns = 0;
    long long current_stagger = 0;
    
    for (long long dmg : a) {
        current_stagger += dmg;
        if (current_stagger >= T) {
            stuns++;
            current_stagger = 0;
        }
    }
    
    return stuns;
}

int main() {
    cout << solve() << endl;
    return 0;
}

File: Day76/Part3/stagger_drain.py
import sys

def solve():
    try:
        input_data = sys.stdin.read().split()
        if not input_data:
            return
        
        iterator = iter(input_data)
        try:
            n = int(next(iterator))
            T = int(next(iterator))
        except StopIteration:
            return
            
        a = []
        for _ in range(n):
            a.append(int(next(iterator)))
            
        stuns = 0
        current_stagger = 0
        
        for dmg in a:
            current_stagger += dmg
            if current_stagger >= T:
                stuns += 1
                current_stagger = 0
                
        print(stuns)
        
    except Exception:
        pass

if __name__ == "__main__":
    solve()
